remotedb/grpcdb/server.go

```diff
@@ -16,11 +16,17 @@ import (
 // ListenAndServe is a blocking function that sets up a gRPC based
 // server at the address supplied, with the gRPC options passed in.
 // Normally in usage, invoke it in a goroutine like you would for http.ListenAndServe.
-func ListenAndServe(addr, cert, key string, opts ...grpc.ServerOption) error {
+func ListenAndServe(addr, cert, key string, opts ...grpc.ServerOption) (rerr error) {
 	ln, err := net.Listen("tcp", addr)
 	if err != nil {
 		return err
 	}
+	defer func() {
```

Comment:
Is there some reason we shouldn't simply defer the close unconditionally? This function blocks until the server terminates anyway, and we're not checking the result of the `Close` call anyway.

Reply:
Yes:
1. It is the simplest and most non-invasive fix, as I've pushed up :-)
2. If at all NewServer returns a value whose .Serve method has goroutines that might do other work on the listener, or also later on invokes .Close on the listener, that could cause problems and the biggest one being an already closed error being reported (something that I've seen in many code bases)

Reply:
The listener is created in this scope, it cannot be closed anywhere else.